module/LearnZF2AjaxImageGallery/src/LearnZF2AjaxImageGallery/Controller/IndexController.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace LearnZF2AjaxImageGallery\Controller;
+
+use LearnZF2AjaxImageGallery\Form\AjaxImageUploadForm;
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\JsonModel;
+use Zend\View\Model\ViewModel;
+use Zend\Json\Json;
+use Admin\Form\ContentForm;
+use Zend\File\Transfer\Adapter\Http;
+use Zend\Validator\File\IsImage;
+use Zend\Validator\File\Size;
+use Zend\Validator\File\Extension;
+
+class IndexController extends AbstractActionController
+{
+    protected $acceptCriteria = [
+        'Zend\View\Model\JsonModel' => ['application/json'],
+        'Zend\View\Model\ViewModel' => ['text/html'],
+    ];
+
+    public $view;
+
+    public function __construct()
+    {
+        $this->view = new ViewModel();

```

Comment:
we need to rely on <code>acceptableviewmodelselector</code> to choose what kind of ViewModel will be used based on accept header.